tensorflow_addons/image/sparse_image_warp.py

```diff
@@ -140,33 +146,38 @@ def sparse_image_warp(
 
 
     Args:
-      image: `[batch, height, width, channels]` float `Tensor`
-      source_control_point_locations: `[batch, num_control_points, 2]` float
-        `Tensor`
-      dest_control_point_locations: `[batch, num_control_points, 2]` float
-        `Tensor`
+      image: Either a 2-D float `Tensor` of shape `[height, width]`,
+        a 3-D `Tensor` of shape `[height, width, channels]`,
+        or a 4-D `Tensor` of shape `[batch_size, height, width, channels]`.
+        `batch_size` is assumed as one when `image` is a 2-D or 3-D `Tensor`.
+      source_control_point_locations: `[batch_size, num_control_points, 2]` float
+        `Tensor`.
+      dest_control_point_locations: `[batch_size, num_control_points, 2]` float
+        `Tensor`.
       interpolation_order: polynomial order used by the spline interpolation
       regularization_weight: weight on smoothness regularizer in interpolation
       num_boundary_points: How many zero-flow boundary points to include at
         each image edge. Usage:
-          `num_boundary_points=0`: don't add zero-flow points
-          `num_boundary_points=1`: 4 corners of the image
-          `num_boundary_points=2`: 4 corners and one in the middle of each edge
-            (8 points total)
-          `num_boundary_points=n`: 4 corners and n-1 along each edge
+        - `num_boundary_points=0`: don't add zero-flow points
+        - `num_boundary_points=1`: 4 corners of the image
+        - `num_boundary_points=2`: 4 corners and one in the middle of each edge
+          (8 points total)
+        - `num_boundary_points=n`: 4 corners and n-1 along each edge
       name: A name for the operation (optional).
 
-      Note that image and offsets can be of type tf.half, tf.float32, or
-      tf.float64, and do not necessarily have to be the same type.
+      Note that `image` and `offsets` can be of type `tf.half`, `tf.float32`, or
+      `tf.float64`, and do not necessarily have to be the same type.
 
     Returns:
-      warped_image: `[batch, height, width, channels]` float `Tensor` with same
-        type as input image.
-      flow_field: `[batch, height, width, 2]` float `Tensor` containing the
+      warped_image: a float `Tensor` with the same shape and dtype as `image`.
+      flow_field: `[batch_size, height, width, 2]` float `Tensor` containing the
```

Comment:
It's better to squeeze the batch dim of `flow_field` when `image` is 2-D or 3-D. The `batch_size` is assumed to be 1 as documentation above. We can deal with it if users request as most of `sparse_image_warp` users want the first output only.